ui/arduino2/store.js

```diff
@@ -524,12 +528,21 @@ async function store(state, emitter) {
   })
 
   // DOWNLOAD AND UPLOAD FILES
+  function confirmDialog(message, cancelText, confirmText) {
+    confirmation = confirm(message, cancelText, confirmText);
+    return confirmation;
+  }
+
```

Comment:
This helper is being defined in the middle of the event handlers. 
While it will work correctly, I would like this to be defined together with the other helper functions, at the end of the file.

Reply:
Also why did you create a helper function for a native function that has exactly the same function signature as the abstraction?

This looks analogous to having a function called `my_alert` where inside you just call `alert` with the exact same parameters of `my_alert`:

```
function my_alert(msg) {
  alert(msg)
}
```

For the final confirmation box we'll do something very different from this so there is no reason to "prepare" for it like this.

Reply:
>Also why did you create a helper function for a native function that has exactly the same function signature as the abstraction?
I thought we wanted to hook into it at some point, that's why I brought it outside.
I'll make the changes later this evening, thanks you